modules/openapi-generator/src/main/resources/python/rest.mustache

```diff
@@ -2,41 +2,67 @@
 
 {{>partial_header}}
 
-import io
 import json
 import logging
 import re
 import ssl
+from typing import Any, Dict, List, Optional, Union, Tuple
 
-from urllib.parse import urlencode, quote_plus
 import urllib3
+from urllib3 import BaseHTTPResponse
+from urllib3.util.timeout import _TYPE_TIMEOUT
 
-from {{packageName}}.exceptions import ApiException, UnauthorizedException, ForbiddenException, NotFoundException, ServiceException, ApiValueError, BadRequestException
+from {{packageName}}.configuration import Configuration
+from {{packageName}}.exceptions import ApiException
+from {{packageName}}.exceptions import ApiValueError
+from {{packageName}}.exceptions import BadRequestException
+from {{packageName}}.exceptions import ForbiddenException
+from {{packageName}}.exceptions import NotFoundException
+from {{packageName}}.exceptions import ServiceException
+from {{packageName}}.exceptions import UnauthorizedException
 
 
 logger = logging.getLogger(__name__)
 
 
-class RESTResponse(io.IOBase):
+class RESTResponse:
+    """An HTTP response."""
+    # This provides a generic object to store HTTP responses.
+    # It proxies the original HTTP response from the underlying HTTP library
+    # (aiohttp, urllib3, etc.) so that clients of RESTClientObject can work
+    # without knowing too much about each library specifics.
 
-    def __init__(self, resp) -> None:
-        self.urllib3_response = resp
+    def __init__(self, resp: BaseHTTPResponse) -> None:
```

Comment:
`BaseHTTPResponse` is only introduced in urllib3 version 2. 

Best use `HTTPResponse`. For me, dropping urllib3 version 1 support would be a problem, because boto3/botocore does not support it yet (https://github.com/boto/botocore/issues/2926).